### PR TITLE
allow incorrect decimal / float formatting

### LINKED ISSUE #12800: Patching non well formed field of type decimal on 3.7.0
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.7.0
* Change in commit: e960656f2a8806f705ec124558963bb1969034ad
* File causing issue: src/Database/Type/DecimalType.php:106

### What you did
Our code is trying to marshall non well formed decimal value (string using comma as decimal separator) into DecimalType database field, which worked up to change in commit mentioned above. 

### What happened
DecimalType method marshall returned null due to removal of !is_scalar($value) condition.

### What you expected to happen
Method should return value if it's scalar as it did in previous version (up to specified commit) as values could be parsed to correct format by entity setters.

### PR DESCRIPTION
**Pull request by segy**
Fixes #12800

### CODE REVIEW

File: src/Database/Type/DecimalType.php [hunk 1]
```diff
@@ -158,7 +158,12 @@ public function marshal($value)
             return (float)$value;
         }
 
-        return null;
+        // allow custom decimal format (@see https://github.com/cakephp/cakephp/issues/12800)
+        if (preg_match('/[^0-9,. ]/', $value)) {
+            return null;
```
Comment (ADmad) [inline src/Database/Type/DecimalType.php]: You have incorrectly swapped the return values :)

File: src/Database/Type/DecimalType.php [hunk 2]
```diff
@@ -158,7 +158,12 @@ public function marshal($value)
             return (float)$value;
         }
 
-        return null;
+        // allow custom decimal format (@see https://github.com/cakephp/cakephp/issues/12800)
```
Comment (ADmad) [inline src/Database/Type/DecimalType.php]: Please remove this comment. You can put the issue link in related test case.

File: src/Database/Type/DecimalType.php [hunk 3]
```diff
@@ -158,7 +158,12 @@ public function marshal($value)
...
+        if (preg_match('/[^0-9,. ]/', $value)) {
```
Comment (ADmad) [inline src/Database/Type/DecimalType.php]: Aren't there some locales which use single/double quotes too as thousands separator?

Comment (dereuromark) [reply src/Database/Type/DecimalType.php]: For degrees lat/lng maybe :) But for decimal I think this is fair enough here.

File: src/Database/Type/DecimalType.php (hunk 1 above)
Comment (dereuromark) [reply src/Database/Type/DecimalType.php]: Sure? The ^ means not afaik.

Comment (segy) [reply src/Database/Type/DecimalType.php]: I hope it's correct

File: src/Database/Type/DecimalType.php (hunk 2 above)
Comment (segy) [reply src/Database/Type/DecimalType.php]: done. moved to test cases

**Comment by segy** [comment]
I wasn't checking the type. Fixed

**Comment by ADmad** [comment]
Nope, the code now allows non scalar values through and tests still fail. We'll save time if you run the tests locally :slightly_smiling_face:.

**Review by ADmad — APPROVED** [review]

**Pull request merged**